### PR TITLE
PDI-12601: Fix for the poor PMR reduce performance with the normal engine reducer

### PR DESCRIPTION
**Pull request by ITXpander**
No more restarts on key groups, streaming as usual

### CODE REVIEW

**Comment by mbatchelor** [comment]
@brosander , @mattyb149 - Please review the PR. Seems like we need a test case on it.

File: build.xml [hunk 1]
```diff
@@ -14,7 +14,7 @@
   <import file="build-res/subfloor-pkg.xml"/>
 
   <!-- Every non-api shim module. These will be compiled/resolved after api. -->
-  <property name="shim-non-api-modules" value="hive-jdbc,hadoop-20,cdh42,cdh50,hdp13,hdp20,hdp21,mapr21,mapr30,mapr31" />
+  <property name="shim-non-api-modules" value="cdh42" />
```
Comment (mattyb149) [inline build.xml]: This change should be reverted, you can override this property from the command-line if you wish

File: common/src-mapred/org/pentaho/hadoop/mapreduce/GenericTransReduce.java [hunk 2]
```diff
@@ -47,329 +40,260 @@
...
-
-        trans = MRUtil.recreateTrans(trans);
-        
-        shareVariableSpaceWithTrans(reporter);
-        setTransLogLevel(reporter);
-        prepareExecution(reporter);
-        addInjectorAndProducerToTrans(key, values, output, reporter, getInputStepName(), getOutputStepName());
+        Reducer<K, V, K2, V2> {
+
+    private static Logger logger = Logger.getLogger(GenericTransReduce.class);
```
Comment (mattyb149) [inline common/src-mapred/org/pentaho/hadoop/mapreduce/GenericTransReduce.java]: This variable (and thus its import) don't appear to be used and are likely not needed/desired

**Comment by mattyb149** [comment]
This looks conspicuously similar to the change for PDI-10729 (BACKLOG-350):

https://github.com/pentaho/pentaho-hadoop-shims/commit/b19abaeed73dcedac8eb5655800d6f8e643e8f00

@lucboudreau @kurtwalker Does this sound familiar? Did you start with this commit then perhaps make additional changes?

@ITXpander please rebase your branch against the latest master, you may find that this (or something very much like it) has been brought in already.

If that's the case, we should dev-test then resolve http://jira.pentaho.com/browse/PDI-12601.  If it is not the case, then a rebase will tell us what remaining changes from this PR to consider.  Thanks!

**Comment by kurtwalker** [comment]
@mattyb149 is correct, we merged this last year.  I think this PR can be closed.

**Pull request closed without merging**